packages/salesforcedx-vscode-core/src/telemetry/telemetry.ts

```diff
@@ -135,7 +136,7 @@ export class TelemetryService {
 
   public dispose(): void {
     if (this.reporter !== undefined) {
-      this.reporter.dispose();
+      this.reporter.dispose().catch();
```

Comment:
Why do we need this `.catch()` during dispose ?

Reply:
Without the `.catch()` I was getting an `UnhandledPromiseRejection` error because this `dispose()` function calls the one declared in `telemetryReporter.ts`. In telemetryReporter's dispose, there is a Promise that expects to be handled, so the `no-floating-promises` linting rule applies here. Alli was helping me figure all this out and we decided to ignore a similar error I was getting in `telemetryReporter.ts` with `// tslint:disable-next-line:no-floating-promises` but kept the `.catch()` here. Is there a better way to handle this scenario, rather than with an empty `catch()`?

Reply:
I see now. You can change it to `.catch(err => {console.log(err)})` or something similar with which we get some visibility if there's an error. I don't think we can do further handling since this is mainly triggered when the editor is closing or the extension is getting deactivated.

Reply:
Okay, sounds good I'll do that!